packages/dashboard-backend/src/routes/api/editors.ts

```diff
@@ -24,7 +26,34 @@ export function registerEditorsRoutes(instance: FastifyInstance) {
     server.get(`${baseApiPath}/editors`, getSchema({ tags }), async () => {
       const token = getServiceAccountToken();
       const { editorsApi } = getDevWorkspaceClient(token);
+
       return editorsApi.list();
     });
+
+    server.get(
+      `${baseApiPath}/editors/devfile`,
+      getSchema({ tags }),
+      async function (request: FastifyRequest, reply: FastifyReply) {
+        const editorId = (request.query as { 'che-editor': string })['che-editor'];
+
+        if (!editorId) {
+          return reply.status(400).send('The che-editor query parameter is required');
+        }
+
+        const token = getServiceAccountToken();
+        const { editorsApi } = getDevWorkspaceClient(token);
+        let editor;
+        try {
+          editor = await editorsApi.get(editorId);
+        } catch (error) {
+          if (error instanceof EditorNotFoundError) {
+            return reply.status(404).send(error);
+          }
+          throw error;
+        }
+
+        return dump(editor);
```

Comment:
I was able to test this code without creating a new helper function

Is it ok @olexii4?

Reply:
It is OK. Thanks.